File: AdvertisementApp.WebApplication/src/models/user/Session.ts
import { ErrorModel } from "../common/ErrorModel";

export enum UserStatus {
  New,
  Registered,
  LoggedIn
}

export enum UserType {
  User,
  Admin
}

export class Session {
  status: UserStatus = UserStatus.New;
  userType: UserType = UserType.User;
  username: string ="";
  success: boolean =false;
  token: string ="";
  error?: ErrorModel;
}

File: AdvertisementApp.BusinessLogic/Models/Users/Session.cs
namespace AdvertisementApp.BusinessLogic.Models.Users
{
    public class Session
    {
        public Session(string userName, bool success, string token, UserType type)
        {
            Username = userName;
            Success = success;
            Token = token;
            UserType = type;
        }

        public string Username { get; set; }
        public bool Success { get; set; }
        public string Token { get; set; }
        public UserType UserType { get; set; }
    }
}